packages/docusaurus-plugin-content-blog/src/readingTime.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+interface ReadingTimeOptions {
+  wordsPerMinute?: number;
+  locale?: string;
+}
+
+interface ReadingTimeResult {
+  text: string;
+  minutes: number;
+  time: number;
+  words: number;
+}
+
+const DEFAULT_WORDS_PER_MINUTE = 200;
+const DEFAULT_LOCALE = 'en';
+
+export function calculateReadingTime(
+  content: string,
+  options: ReadingTimeOptions = {},
+): ReadingTimeResult {
+  const wordsPerMinute = options.wordsPerMinute ?? DEFAULT_WORDS_PER_MINUTE;
+  const locale = options.locale ?? DEFAULT_LOCALE;
+  const contentWithoutFrontmatter = content.replace(/^---[\s\S]*?---\n/, '');
```

Comment:
We didn't have that before so I'd prefer to not do that.

The called should be responsible from providing text content, and this function shouldn't assume it's called in a markdown/mdx context